io/src/lib.rs

```diff
@@ -115,6 +170,22 @@ impl<T: AsRef<[u8]>> Read for Cursor<T> {
     }
 }
 
+impl<T: AsRef<[u8]>> BufRead for Cursor<T> {
+    #[inline]
+    fn fill_buf(&mut self) -> Result<&[u8]> {
+        let inner: &[u8] = self.inner.as_ref();
+        Ok(&inner[self.pos as usize..])
+    }
+
+    #[inline]
+    fn consume(&mut self, amount: usize) {
+        // Don't let callers reset the limit by passing an overlarge value.
+        let amount = cmp::min(amount as u64, self.remaining) as usize;
```

Comment:
Why not just compare `self.inner.as_ref().len()`? Sure if the callers do something stupid like implementing `AsRef<[u8]>` to return different slices they will get problems but that applies anyway.

Reply:
Used an `assert!` instead.